src/logic/basic.lean

```diff
@@ -82,16 +82,26 @@ lemma subtype.subsingleton (α : Sort*) [subsingleton α] (p : α → Prop) : su
   (a : α) : (a : γ) = (a : β) := rfl
 
 theorem coe_fn_coe_trans
-  {α β γ} [has_coe α β] [has_coe_t_aux β γ] [has_coe_to_fun γ]
-  (x : α) : @coe_fn α _ x = @coe_fn β _ x := rfl
+  {α β γ δ} [has_coe α β] [has_coe_t_aux β γ] [has_coe_to_fun γ δ]
+  (x : α) : @coe_fn α _ _ x = @coe_fn β _ _ x := rfl
+
+/-- Non-dependent version of `coe_fn_coe_trans`, helps `rw` figure out the argument. -/
+theorem coe_fn_coe_trans'
+  {α β γ} {δ : out_param $ _} [has_coe α β] [has_coe_t_aux β γ] [has_coe_to_fun γ (λ _, δ)]
+  (x : α) : @coe_fn α _ _ x = @coe_fn β _ _ x := rfl
 
 @[simp] theorem coe_fn_coe_base
-  {α β} [has_coe α β] [has_coe_to_fun β]
-  (x : α) : @coe_fn α _ x = @coe_fn β _ x := rfl
+  {α β γ} [has_coe α β] [has_coe_to_fun β γ]
+  (x : α) : @coe_fn α _ _ x = @coe_fn β _ _ x := rfl
+
+/-- Non-dependent version of `coe_fn_coe_base`, helps `rw` figure out the argument. -/
+theorem coe_fn_coe_base'
```

Comment:
I would like to tag this as `@[simp]`, but that causes a timeout in `src/order/rel_iso.lean:465` which I don't quite know how to diagnose.